var/spack/repos/builtin/packages/pigz/package.py

```diff
@@ -20,7 +20,7 @@ class Pigz(MakefilePackage):
     depends_on('zlib')
 
     def build(self, spec, prefix):
-        make()
+        make('CC=%s' % os.environ['CC'])
```

Comment:
```suggestion
        make('CC=%s' % self.compiler.cc)
```

Reply:
The trouble with using self.compiler.cc is that if you are using ccache to compile self.compiler.cc may be pointing to the local ccache executable which will cause a failure.  The CC environment variable will point to the relative path of CC.

Reply:
Not sure if I follow:

```
>>> spack.spec.Spec('zlib%gcc').concretized().package.compiler.cc
'/usr/bin/gcc'
>>> spack.config.get('config:ccache')
True
```

Reply:
Yeah CC=cc makes a lot of sense  Let me fix the patch

The issue wasnt ccache, but distcc

 >> 79    make: *** [Makefile:37: cache.o] Error 111
     80    distcc[434] (main) CRITICAL! distcc seems to have invoked itself rec
           ursively!
     81    distcc[414] ERROR: compile zopfli/src/zopfli/tree.c on localhost fai
           led with exit code 111
  >> 82    make: *** [Makefile:31: tree.o] Error 111
     83    distcc[436] (main) CRITICAL! distcc seems to have invoked itself rec
           ursively!
     84    distcc[421] ERROR: compile zopfli/src/zopfli/squeeze.c on localhost 
           failed with exit code 111
  >> 85    make: *** [Makefile:46: squeeze.o] Error 111
     86    distcc[438] (main) CRITICAL! distcc seems to have invoked itself rec
           ursively!
     87    distcc[419] ERROR: compile zopfli/src/zopfli/util.c on localhost fai
           led with exit code 111
  >> 88    make: *** [Makefile:43: util.o] Error 111
     89    distcc[439] (main) CRITICAL! distcc seems to have invoked itself rec
           ursively!
     90    distcc[422] ERROR: compile zopfli/src/zopfli/katajainen.c on localho
           st failed with exit code 111
  >> 91    make: *** [Makefile:49: katajainen.o] Error 111
     92    distcc[437] (main) CRITICAL! distcc seems to have invoked itself rec
           ursively!
     93    distcc[418] ERROR: compile zopfli/src/zopfli/hash.c on localhost fai
           led with exit code 111
  >> 94    make: *** [Makefile:40: hash.o] Error 111

Reply:
Right, actually the correct magic variable is `'CC=%s' % spack_cc`, it's defined by Spack. But `CC=cc` is shorter... so both are fine for me.